packages/angular_devkit/core/src/workspace/json/reader.ts

```diff
@@ -218,7 +218,7 @@ function parseProject(
         break;
       default:
         if (!context.unprefixedProjectExtensions.has(name) && !/^[a-z]{1,3}-.*/.test(name)) {
-          context.warn(`Project extension with invalid name (${name}) found.`, name);
+          context.warn(`Project ${projectName} contains extension with invalid name (${name}) found.`, name);
```

Comment:
This is now grammatically incorrect, in particular the "found" at the end.

```suggestion
          context.warn(`Project '${projectName}' contains extension with invalid name (${name}).`, name);
```

the surrounding quotes are a suggestion to make it more apparant that the message contains a project name.